onnxruntime/core/providers/mkldnn/subgraph/mkldnn_activations.h

```diff
@@ -44,25 +44,27 @@ class MklDnnRelu : public MklDnnKernel {
       mkldnn::memory::dims dims(xdim);
 
       ort_source_format_ = GetSourceFormat(static_cast<int>(xdim));
-      source_format = ort_source_format_;
-      src_format_ = ort_source_format_;
 
-      x_shape = TensorShape(xshape, xdim);
+	  x_shape = TensorShape(xshape, xdim);
 
-      mkldnn::memory::dims src_dims_mkl(
+      mkldnn::memory::dims src_dims(
           x_shape.GetDims().begin(), x_shape.GetDims().end());
 
+      ort_source_desc_ = mkldnn::memory::desc(
+          {src_dims}, MklDnnType<T>(), ort_source_format_);
+      source_desc_ = ort_source_desc_;
       src_md_.reset(new mkldnn::memory::desc(
```

Comment:
we can take this opportunity to clean up the smart pointer usage in all these files.
src_md_ = make_shared<T>(...) is preferred and more efficient than reset(new T)
let's clean all this up. thanks.

Reply:
Most of the pointers are unique_ptr and we need make_unique.
I was about to push my changes. I will make these changes and then create a new PR

Reply:
> Most of the pointers are unique_ptr and we need make_unique.
> I was about to push my changes. I will make these changes and then create a new PR

let's do the cleanup as a separate PR then. priority is to get this one passing the CI's and merged soon.

Reply:
okay. I will close this and create new new PR. As I have merged latest from master repo.